content/3_Silver/Sorting_Custom.mdx

```diff
@@ -48,626 +46,461 @@ After sorting, it should look like
-object $y$ in a sorted ordering.
+Essentially, the comparator determines whether $x$ belongs to the left of
+$y$ in a sorted ordering.
 
 <Warning>
 
-A comparator **must** return false for two equal objects (not doing so results
-in undefined behavior and potentially a verdict of wrong answer or runtime
-error).
+A comparator **must** return false for two equal objects.
+Not doing so
+[results in undefined behavior](https://stackoverflow.com/a/45929812/12128483).
 
 </Warning>
 
-In addition to returning the correct answer, comparators should also satisfy the
-following conditions:
-
-- The function must be consistent with respect to reversing the order of the
-  arguments: if $x \neq y$ and `compare(x, y)` is `true`, then `compare(y, x)`
-  should be `false` and vice versa.
-- The function must be transitive. If `compare(x, y)` is true and
-  `compare(y, z)` is true, then `compare(x, z)` should also be true. If the
-  first two compare functions both return `false`, the third must also return
-  `false`.
-
-### Method 1 - Overloading the Less Than Operator
+### [Overloading](https://en.cppreference.com/w/cpp/language/operators) `operator<`
 
 This is the easiest to implement. However, it only works for objects (not
 primitives) and it doesn't allow you to define multiple ways to compare the same
 type of class.
 
-In the context of Wormhole Sort (note the use of
-[const Edge&](https://stackoverflow.com/questions/11805322/why-should-i-use-const-t-instead-of-const-t-or-t)):
+Let's sort those four edges again with this new method:
 
 ```cpp
-#include <bits/stdc++.h>
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
 using namespace std;
 
 struct Edge {
-	int a, b, w;
-	bool operator<(const Edge &y) { return w < y.w; }
+	int a, b;
+	int width;
+
+	// see https://stackoverflow.com/a/11805395/12128483 for why we use const Edge&
+	bool operator<(const Edge &y) { return width < y.width; }
 };
 
 int main() {
-	int M = 4;
-	vector<Edge> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({a, b, w});
-	}
-	sort(begin(v), end(v));
-	for (Edge e : v) cout << e.a << " " << e.b << " " << e.w << "\n";
+	constexpr int edge_num = 4;
+
+	vector<Edge> edges(edge_num);
+	for (Edge &e : edges) { cin >> e.a >> e.b >> e.width; }
+	sort(edges.begin(), edges.end());
+
+	for (const Edge &e : edges) { printf("(%i, %i): %i\n", e.a, e.b, e.width); }
 }
 ```
 
-We can also overload the operator outside of the class:
+It's also possible to define `operator<` outside of the class:
 
 ```cpp
 struct Edge {
-	int a, b, w;
+	int a, b;
+	int width;
 };
-bool operator<(const Edge &x, const Edge &y) { return x.w < y.w; }
+
+bool operator<(const Edge &x, const Edge &y) { return x.width < y.width; }
 ```
 
 or within it using
-[friend](https://www.geeksforgeeks.org/friend-class-function-cpp/):
+[`friend`](https://en.cppreference.com/w/cpp/language/friend):
```

Comment:
hm i mean im not sure that there's enough difference between just defining it and using friend. from what i know:
* all struct fields are public by default
* friend gives access to private/protected members

might be worth clarifying/removing?

Reply:
the og module talks about it
yeah i agree there's not much point in having this tho, we can remove it

Reply:
yeah isn't defining it inside the struct or writing a separate comparator always more practical